lib/optimize/MergeDuplicateChunksPlugin.js

```diff
@@ -2,29 +2,32 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-function MergeDuplicateChunksPlugin() {}
-module.exports = MergeDuplicateChunksPlugin;
+"use strict";
 
 function getChunkIdentifier(chunk) {
-	return chunk.modules.map(function(m) {
+	return chunk.modules.map((m) => {
 		return m.identifier();
 	}).sort().join(", ");
 }
 
-MergeDuplicateChunksPlugin.prototype.apply = function(compiler) {
-	compiler.plugin("compilation", function(compilation) {
-		compilation.plugin("optimize-chunks-basic", function(chunks) {
-			var map = {};
-			chunks.slice().forEach(function(chunk) {
-				if(chunk.hasRuntime() || chunk.hasEntryModule()) return;
-				var ident = getChunkIdentifier(chunk);
-				if(map[ident]) {
-					if(map[ident].integrate(chunk, "duplicate"))
-						chunks.splice(chunks.indexOf(chunk), 1);
-					return;
-				}
-				map[ident] = chunk;
+class MergeDuplicateChunksPlugin {
+
+	apply(compiler) {
+		compiler.plugin("compilation", (compilation) => {
+			compilation.plugin("optimize-chunks-basic", (chunks) => {
+				const map = {};
```

Comment:
This would be a perfect place to replace the object with `new Map()` and instead of `if(map[ident])` I believe Map has a `.has` fn you can use to test if it exists or not.